stix_shifter_modules/cisco_secure_email/stix_transmission/connector.py

```diff
@@ -0,0 +1,167 @@
+                    result_offset += len(response_dict['data'])
+                elif response_wrapper.code == 401 and response_dict['error']['message'] in ['ExpiredSignatureError.',
+                                                                                            'InvalidTokenError.']:
+                    jwt_token = await self.get_token()
+                    continue
+                else:
+                    return_obj = self.exception_response(response_wrapper.code, response_dict['error'])
+                    data = []
+                    break
+            if data:
+                data = self.add_is_multipart(data)
+                if metadata:
+                    return_obj['data'] = data if data else []
+                else:
+                    return_obj['data'] = data[int(offset): total_records] if data else []
+                if jwt_token:
+                    return_obj['metadata'] = {"jwtToken": jwt_token}
+            else:
+                if not return_obj.get('error') and return_obj.get('success') is not False:
+                    return_obj['success'] = True
+                    return_obj['data'] = []
+
+        except InvalidMetadataException as ex:
+            response_dict['code'] = 100
+            response_dict['message'] = f'Invalid metadata: {str(ex)}'
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except Exception as ex:
+            if "server timeout_error" in str(ex):
+                response_dict['code'] = 503
+            elif "timeout_error" in str(ex):
+                response_dict['code'] = 408
+            response_dict['message'] = str(ex)
+            self.logger.error('error while fetching results: %s', ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        return return_obj
+
+    async def ping_connection(self):
+        """
+        Ping the endpoint
+        :return: return_object, dict
+        """
+        return_obj = {}
+        response_dict = {}
+        try:
+            token = await self.get_token()
+            response = await self.api_client.ping_data_source(token)
+            response_code = response.code
+            response_dict = json.loads(response.read().decode('utf-8'))
+            if response_code == 200:
+                return_obj['success'] = True
+            else:
+                return_obj = self.exception_response(response_code, response_dict.get('errorSummary', ''))
+
+        except Exception as ex:
+            if "server timeout_error" in str(ex):
+                response_dict['code'] = 503
+            elif "timeout_error" in str(ex):
+                response_dict['code'] = 408
+            response_dict['message'] = str(ex)
+            self.logger.error('error while pinging: %s', ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
```

Comment:
you need to handle SSLError as well. i have got an error with `unknown` error code when i put invalid certificate.

```
{
    "success": false,
    "connector": "cisco_secure_email",
    "error": "cisco_secure_email connector error => [X509] PEM lib (_ssl.c:<code>)",
    "code": "unknown"
}
```

Reply:
Updated code to handle Invalid certificate exception.